defaults/genesis.json

```diff
@@ -59,7 +59,7 @@
       "Register": {
         "AssetDefinition": {
           "id": "rose#wonderland",
-          "type": "Numeric",
+          "spec": "Numeric",
```

Comment:
This seemingly doesn't make sense. `spec` must be `{ "scale": 0 }` or something, right?

Reply:
Currently we have this behaviour for formatting `NumericSpec`:
* `scale = 0` is formatted as `Numeric(0)`
* `scale = None` is formatted as just `Numeric`

https://github.com/hyperledger-iroha/iroha/blob/591dd5e967a2f8fe6db37e4b5d2eda8484254292/crates/iroha_numeric/src/lib.rs#L380-L385

---

One option I see here is to change `spec: NumericSpec` field of `AssetDefinition` to just `scale: Option<u32>`

Reply:
> This seemingly doesn't make sense. `spec` must be `{ "scale": 0 }` or something, right?

No, it's not necessary that spec is defined. When scale is undefined the number is a variable precision as opposed to fixed precision when it is defined

Reply:
> Currently we have this behaviour for formatting `NumericSpec`:
> 
>     * `scale = 0` is formatted as `Numeric(0)`
> 
>     * `scale = None` is formatted as just `Numeric`

Oh, I see, I didn't know that. This is still a confusing thing though. I understand that it looks readable in some cases, but it is not consistent with SCALE representation, and is not documented anywhere. I would advocate for not having exotic custom-parsed strings in the JSON repr, and use `{ scale: Option<u32> }`.

> One option I see here is to change `spec: NumericSpec` field of `AssetDefinition` to just `scale: Option<u32>`

Fine too.